addons/binding/org.openhab.binding.facerecognition/src/main/java/org/openhab/binding/facerecognition/handler/FaceRecognitionHandler.java

```diff
@@ -0,0 +1,310 @@
+                } else {
+                    logger.debug("No faces trained yet");
+
+                    // Set default unknown user values
+                    label.put(UNKNOWN_USER_LABEL);
+                    score.put(Double.MAX_VALUE);
+                }
+
+                // User recognized?
+                if (facesTrained && (score.get(0) <= threshold.intValue())) {
+
+                    if (isLinked(CHANNEL_ACCESS_GRANTED_IMAGE)) {
+                        ByteBuffer outputBuffer = ByteBuffer.allocate(face.arraySize());
+                        imencode(".jpg", face, outputBuffer);
+                        updateState(accessGrantedImageChannelUID, new RawType(outputBuffer.array(), "image/jpeg"));
+                    }
+
+                    if (isLinked(CHANNEL_ACCESS_GRANTED_USERNAME)) {
+                        updateState(accessGrantedUsernameChannelUID,
+                                new StringType(faceRecognizer.getLabelInfo(label.get(0)).getString()));
+                    }
+
+                    if (isLinked(CHANNEL_ACCESS_GRANTED_SCORE)) {
+                        updateState(accessGrantedScoreChannelUID, new DecimalType(score.get(0)));
+                    }
+
+                    if (isLinked(CHANNEL_ACCESS_GRANTED_TIME)) {
+                        updateState(accessGrantedTimeChannelUID, new DateTimeType());
+                    }
+
+                } else {
+
+                    if (isLinked(CHANNEL_ACCESS_DENIED_IMAGE)) {
+                        accessDeniedFace = new Mat(face);
+                        ByteBuffer outputBuffer = ByteBuffer.allocate(accessDeniedFace.arraySize());
+                        imencode(".jpg", accessDeniedFace, outputBuffer);
+                        updateState(accessDeniedImageChannelUID, new RawType(outputBuffer.array(), "image/jpeg"));
+                    }
+
+                    if (isLinked(CHANNEL_ACCESS_DENIED_USERNAME)) {
+                        String user;
+
+                        // User not known?
+                        if (label.get(0) == UNKNOWN_USER_LABEL) {
+                            user = UNKNOWN_USER_NAME;
+                        } else {
+                            user = faceRecognizer.getLabelInfo(label.get(0)).getString();
+                        }
+
+                        updateState(accessDeniedUsernameChannelUID, new StringType(user));
+                    }
+
+                    if (isLinked(CHANNEL_ACCESS_DENIED_SCORE)) {
+                        updateState(accessDeniedScoreChannelUID, new DecimalType(score.get(0)));
+                    }
+
+                    if (isLinked(CHANNEL_ACCESS_DENIED_TIME)) {
+                        updateState(accessDeniedTimeChannelUID, new DateTimeType());
+                    }
+                }
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+
+        } catch (Exception e) {
+            logger.error("{}: {}", e.getClass().getName(), e.getMessage());
```

Comment:
logger.error. That's also not necessary infront of `updateStatus`. And you are catching a generic exception here.